app/views.py

```diff
@@ -84,6 +94,69 @@ def get_methods():
 def get_method(method):
     return jsonize(models.get_method(obfuscated_id=as_id(method)))
 
+
+@app.route('/prepare-multipart-upload', methods=['POST'])
+def prepare_file_upload():
+    pass
+
+
+@app.route('/upload-part', methods=['POST'])
+def upload_file():
+    part = request.files['file']
+
+    # part_number = request.form['part-number']
+    part_number = request.form['resumableChunkNumber']
+    # total_number_parts = request.form['total-parts']
+    total_number_parts = request.form['resumableTotalChunks']
+
+    # file_identifier = request.form['upload-id']
+    file_identifier = request.form['resumableIdentifier']
+    part_upload_dir = upload_dir(file_identifier)
+    mkdirp(part_upload_dir)
+
+    # Save the part into a file with a name of the form 00.part.  The number of
+    # leading zeroes depends on the number of parts.
+    fmtstr = "%%0%dd" % len(total_number_parts)
+    part_filename = (fmtstr % int(part_number)) + PART_EXT
+    part.save(os.path.join(part_upload_dir, part_filename))
+
+    return json.dumps({'identifier': file_identifier,
+                       'part': part_number,
+                       'total_parts': total_number_parts})
+
+
+@app.route('/complete-multipart-upload', methods=['POST'])
+def complete_file_upload():
+    request_data    = json.loads(request.data)
+    file_identifier = request_data['upload-id']
+    file_name       = secure_filename(request_data['file-name'])
+    project         = request_data['project']
+    sample          = request_data['sample']
+    sample_stage    = as_id(request_data['sample-stage'])
+
+    # TODO: Client must include a file hash!
+
+    part_upload_dir = upload_dir(file_identifier)
+    reconstituted_file_name = os.path.join(part_upload_dir, file_name)
+    parts = glob.glob(os.path.join(part_upload_dir, '*.part'))
```

Comment:
Is this guaranteed to return the parts in lexicographical order? Does this return an iterable? If so, 'list(sorted(glob.glob(...)))' will guarantee it is sorted. We may not the 'list(...)' construction, but I can't remember if sorted returns a list or a generator.